contracts/src/contracts/Teleporter/AvalancheICTTRouterLooseFees.sol

```diff
@@ -0,0 +1,199 @@
+contract AvalancheICTTRouterLooseFees is Ownable, ReentrancyGuard, IAvalancheICTTRouterLooseFees {
+    using Address for address;
+
+    /**
+     * @notice Token address => source bridge address
+     * @notice Address `0x0` is used for the native token
+     */
+    mapping(address token => address sourceBridge) public tokenToSourceBridge;
+
+    /**
+     * @notice Token address => destination chain ID => DestinationBridge
+     * @notice Address `0x0` is used for the native token
+     */
+    mapping(
+        bytes32 destinationChainID => mapping(address token => DestinationBridge destinationBridge)
+    ) public tokenDestinationChainToDestinationBridge;
+
+    /// @notice  Current chain ID
+    bytes32 private immutable routerChainID;
+
+    /// @notice Set the relayer fee and the ID of the source chain
+    constructor() {
+        routerChainID = IWarpMessenger(0x0200000000000000000000000000000000000005).getBlockchainID();
+    }
+
+    function registerSourceTokenBridge(
+        address tokenAddress,
+        address bridgeAddress
+    ) external onlyOwner {
+        if (tokenAddress != address(0) && !tokenAddress.isContract()) {
+            revert NotAContract(tokenAddress);
+        }
+        if (!bridgeAddress.isContract()) {
+            revert NotAContract(bridgeAddress);
+        }
+        tokenToSourceBridge[tokenAddress] = bridgeAddress;
+
+        emit RegisterSourceTokenBridge(tokenAddress, bridgeAddress);
+    }
+
+    function registerDestinationTokenBridge(
+        address tokenAddress,
+        bytes32 destinationChainID,
+        address bridgeAddress,
+        uint256 requiredGasLimit,
+        bool isMultihop
+    ) external onlyOwner {
+        if (tokenAddress != address(0) && !tokenAddress.isContract()) {
+            revert NotAContract(tokenAddress);
+        }
+        if (bridgeAddress == address(0)) {
+            revert NotAContract(bridgeAddress);
+        }
+        if (destinationChainID == routerChainID) {
+            revert SourceChainEqualToDestinationChain(routerChainID, destinationChainID);
+        }
+        DestinationBridge memory destinationBridge =
+            DestinationBridge(bridgeAddress, requiredGasLimit, isMultihop);
+        tokenDestinationChainToDestinationBridge[destinationChainID][tokenAddress] =
+            destinationBridge;
+
+        emit RegisterDestinationTokenBridge(tokenAddress, destinationBridge, destinationChainID);
+    }
+
+    function removeSourceTokenBridge(address tokenAddress) external onlyOwner {
+        delete tokenToSourceBridge[tokenAddress];
+
+        emit RemoveSourceTokenBridge(tokenAddress);
+    }
+
+    function removeDestinationTokenBridge(
+        address tokenAddress,
+        bytes32 destinationChainID
+    ) external onlyOwner {
+        delete tokenDestinationChainToDestinationBridge[destinationChainID][tokenAddress];
+
+        emit RemoveDestinationTokenBridge(tokenAddress, destinationChainID);
+    }
```

Comment:
Why duplicate all those functions? Implement them in the `AvalancheICTTRouter` and make `AvalancheICTTRouterFixedFees` inherit it + override the 2 functions that shouldn't be called.